lib/timeutils/unixtime.c

```diff
@@ -45,6 +47,133 @@ unix_time_set_now(UnixTime *self)
   self->ut_gmtoff = get_local_timezone_ofs(self->ut_sec);
 }
 
+static glong
+_div_round(glong n, glong d)
```

Comment:
Just a minor thing, but I think it would be more clear to abstract out a `_round(number, base)` function instead: the multiplication should be part of it.

This function is a special version of division in a way that after multiplying we would get the correct result in terms of rounding. But this function cannot stand on its own, or at least I cannot see its usefulness in another context.

Reply:
This is a division that rounds the result according to the mathematical definition, e.g. one that rounds upwards above .5 and downwards below that.

5/2 = 3
5/3 = 2
5/4 = 1

With non-floating point division, 5/2 would be 2. The name of the function could be improved somewhat, and you might be right that we should include the multiplication, let me think if there's a good function name for that...

Reply:
I gave it a thought and I think the current name is not ultimately bad. It is a function that does division rounding the result. With that i resolve this conversation

Reply:
I agree the function name is fine.

I wanted to imply that wrong function was extracted. Rounding has a strict mathematical formula, but the patch does not contain function for that. Instead a subexpression from the formula was extracted into the function. I think it would be better extracting the rounding itself into the function, instead of extracting _div_round. (Or both could be extracted.)

Still, this is just a minor thing. It is fine for me if this stays as it is.